### ISSUE TITLE
L003: Allow comment lines to be ignored

### ISSUE THREAD

**Issue by jivanpal** [sqlfluff/sqlfluff#3311]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

I notice that L003 reports different messages for comments and for code:

- "Comment not aligned with following line." *(Aside: Should really say "preceding line", no?)*
- "Expected X indentations, found Y [compared to line N]"

So presumably, it wouldn't be too difficult to allow it to ignore comments completely (i.e. never report messages of the first kind), perhaps via the existing [`ignore_comment_lines` parameter used by L016](https://docs.sqlfluff.com/en/stable/rules.html#sqlfluff.core.rules.Rule_L016).

See use case below.

### Use case

I frequently write schemas with inline documentation, e.g.

```sql
CREATE TABLE t (
    a INT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT,
    b VARCHAR(63),
        -- Description of what `b` is for
    c BOOLEAN NOT NULL
        /* Long description of
        what `c` is for */
);
``` 

Currently, L003 complains about the comments as they are indented further than the line that precedes them, but I would like this not to happen, i.e. would like the above to be accepted without disabling L003 in its entirety.

### Dialect

Shouldn't be dialect-specific, but I use `mysql`.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Nate-Wessel** [comment]
On a related note, I'm also seeing some issues where SQLFluff `fix` is reformatting indentation within long block comments. Is there a preferred way of indenting within comments? That seems like a stretch to me. I would prefer to be able to ignore the contents of block comments entirely.

**Comment by alanmcruickshank** [comment]
@Nate-Wessel - I think this should be at least partly resolved in the most recent alpha releases: https://github.com/sqlfluff/sqlfluff/issues/4224

More generally here though, there is a question on whether we should enforce _any_ specific indentation within block comments - so far we just make sure things are _at least as indented_ as the first line, but otherwise be fairly permissive.

As for the original post here (@jivanpal), we're currently assuming the convention that the documentation comment for field is expected on the line _before_ the line it refers to rather than _after_ and with the _same indent_. This is broadly in line with sql style guides (like the [dbt style guide](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md)) and in line with other language conventions (like [python](https://peps.python.org/pep-0008/)).

With that style in mind, the indentation which currently passes the indentation rules on the current `main` branch is:
```sql
CREATE TABLE t (
    a INT UNSIGNED NOT NULL PRIMARY KEY AUTO_INCREMENT,
    -- Description of what `b` is for
    b VARCHAR(63),
    /* Long description of
    what `c` is for */
    c BOOLEAN NOT NULL
);
```
